components/experimental/src/units/converter_ratio.rs

```diff
@@ -0,0 +1,56 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::units::ratio::IcuRatio;
+
+// TODO: add Mul & Add for references to avoid cloning.
+/// A trait for types that can be converted between two units.
+pub trait ConverterRatio: Clone {
+    fn reciprocal(&self) -> Self;
+
+    // TODO: remove this method once implement Mul & Add for references to avoid cloning.
+    fn add(&self, other: &Self) -> Self;
+
+    // TODO: remove this method once implement Mul & Add for references to avoid cloning.
+    fn mul(&self, other: &Self) -> Self;
+
+    // TODO: is there a way to implement this for all types that implement From<IcuRatio>?
+    fn from_icu_ratio(value: IcuRatio) -> Option<Self>;
+}
+
+impl ConverterRatio for IcuRatio {
+    fn reciprocal(&self) -> Self {
+        self.recip()
+    }
+
+    fn from_icu_ratio(value: IcuRatio) -> Option<Self> {
+        Some(value)
+    }
+
+    fn mul(&self, other: &Self) -> Self {
+        self * other
+    }
+
+    fn add(&self, other: &Self) -> Self {
+        self + other
+    }
+}
+
+impl ConverterRatio for f64 {
```

Comment:
issue: `f64` is not a ratio, the trait name is not general enough

Reply:
Agreed

Reply:
shall we call it `Convertable` ?

Reply:
Let's see what's left after applying Manish's suggestion to use the traits from `num_traits`.